contracts/SpeedMarkets/MultiCollateralOnOffRamp.sol

```diff
@@ -112,13 +117,48 @@ contract MultiCollateralOnOffRamp is Initializable, ProxyOwned, ProxyPausable, P
         uint balanceDiff = IERC20Upgradeable(WETH9).balanceOf(address(this)) - balanceBefore;
         require(balanceDiff == amount, "Not enough WETH received");
 
-        convertedAmount = _swapExactSingle(amount, WETH9);
+        convertedAmount = _swapExactInput(amount, WETH9);
 
         sUSD.safeTransfer(msg.sender, convertedAmount);
 
         emit OnrampedEth(amount);
     }
 
+    /// @notice offramp the amount of sUSD into the target collateral
+    /// @return convertedAmount The amount of sUSD to offramp.
+    function offramp(address collateral, uint amount) external nonReentrant notPaused returns (uint convertedAmount) {
+        require(collateralSupported[collateral], "Unsupported collateral");
+        require(ammsSupported[msg.sender], "Unsupported caller");
+        sUSD.safeTransferFrom(msg.sender, address(this), amount);
+
+        if (curveOnrampEnabled && (collateral == usdc || collateral == dai || collateral == usdt)) {
+            // for stable coins use Curve
+            convertedAmount = _swapViaCurveOfframp(collateral, amount);
+        } else {
+            // for other use defined path
+            convertedAmount = _swapExactInputOfframp(amount, collateral);
+        }
+        IERC20Upgradeable(collateral).safeTransfer(msg.sender, convertedAmount);
+
+        emit Offramped(collateral, amount);
+    }
+
+    function offrampIntoEth(uint amount) external nonReentrant notPaused returns (uint convertedAmount) {
+        require(ammsSupported[msg.sender], "Unsupported caller");
+
+        sUSD.safeTransferFrom(msg.sender, address(this), amount);
+
+        convertedAmount = _swapExactInputOfframp(amount, WETH9);
+
+        WethLike(WETH9).withdraw(convertedAmount);
+
+        address payable _to = payable(msg.sender);
+        bool sent = _to.send(convertedAmount);
```

Comment:
line 156 without 155 is sufficient
`bool sent = payable(msg.sender).send(convertedAmount)`

Reply:
changed